packages/api/src/promise/types.d.ts

```diff
@@ -7,15 +7,15 @@ import { ApiBaseInterface } from '../types';
 import { Observable } from 'rxjs';
 import { ExtrinsicFunction } from '@polkadot/extrinsics/types';
 import Rpc from '@polkadot/rpc-core/index';
-import { Base } from '@polkadot/types/codec';
+import { Codec } from '@polkadot/types/codec';
```

Comment:
I am pretty sure this line does not work. But importing from '@polkadot/types' (to get the Codec interface from types.d) does not seem to work either. 
What can be imported here?

Furthermore: import { Base } from '@polkadot/types/codec' occurs 2 times more. 
One within rx/types.d.ts and another in api/types.d.ts

Reply:
I have no idea why the compiler doesn't throw an error here, it definitely should...

Anyways, the correct import should be `import { Codec } from '@polkadot/types/types`, which do a couple of times already.

And yes, the 2 other occurrences should be replaced too.

Reply:
Yeap, passes on CI. This is beyond weird :(

Reply:
Thanks. I did as seen below. Test and build is running successfully - but I have not checked the build process in detail to know why the compiler does not complain.

Reply:
https://github.com/polkadot-js/api/issues/406